src/core/ext/client_channel/client_channel.c

```diff
@@ -546,10 +546,14 @@ static grpc_error *cc_init_channel_elem(grpc_exec_ctx *exec_ctx,
   arg = grpc_channel_args_find(args->channel_args, GRPC_ARG_SERVER_URI);
   GPR_ASSERT(arg != NULL);
   GPR_ASSERT(arg->type == GRPC_ARG_STRING);
-  chand->server_name = gpr_strdup(arg->value.string);
+  grpc_uri *uri = grpc_uri_parse(arg->value.string, true);
+  GPR_ASSERT(uri->path[0] != '\0');
```

Comment:
this will crash if `uri` is NULL, which it could be if the `target` passed to `client_channel_factory_create_channel` is NULL. While the current code would actually crash even earlier than that (`resolve_factory` as called from `grpc_resolver_factory_add_default_prefix_if_needed` assumes `target` is non-NULL) I think it's still worth checking that the result of the URI parse is corrrect.

In fact we may want to validate `target` as early as `client_channel_factory_create_channel` so that we can work with the assumption that is isn't NULL further down the stack.